pkg/cache/application_state.go

```diff
@@ -31,76 +31,102 @@ import (
 // ----------------------------------
 // application events
 // ----------------------------------
-type ApplicationEvent int
+type applicationEvent int
 
 const (
-	AcceptApplication ApplicationEvent = iota
-	RejectApplication
+	AcceptApplication applicationEvent = iota
+	StartApplication
 	RunApplication
+	WaitApplication
+	RejectApplication
 	CompleteApplication
 	KillApplication
 )
 
-func (ae ApplicationEvent) String() string {
-	return [...]string{"AcceptApplication", "RejectApplication", "RunApplication", "CompleteApplication", "KillApplication"}[ae]
+func (ae applicationEvent) String() string {
+	return [...]string{"AcceptApplication", "StartApplication", "RunApplication", "WaitApplication", "RejectApplication", "CompleteApplication", "KillApplication"}[ae]
 }
 
 // ----------------------------------
 // application states
 // ----------------------------------
-type ApplicationState int
+type applicationState int
 
 const (
-	New ApplicationState = iota
+	New applicationState = iota
 	Accepted
-	Rejected
+	Starting
 	Running
+	Waiting
+	Rejected
 	Completed
 	Killed
```

Comment:
I found the meaning of these states is really confusing.
Can you please add the comment here, and also in a doc. I think we need a design doc for this as well.

Reply:
First doc info added to the jira, need to write a full doc for it, including graphs.
We need to fully document all states that we have in the system.